mobile/src/main/java/com/gelakinetic/mtgfam/helpers/CardHelpers.java

```diff
@@ -628,125 +610,115 @@ public int compare(CompressedDecklistInfo card1, CompressedDecklistInfo card2) {
-                        .getColumnIndex(CardDbAdapter.KEY_MANACOST));
-                card.mPower = cardCursor.getInt(cardCursor
-                        .getColumnIndex(CardDbAdapter.KEY_POWER));
-                card.mToughness = cardCursor.getInt(cardCursor
-                        .getColumnIndex(CardDbAdapter.KEY_TOUGHNESS));
-                card.mLoyalty = cardCursor.getInt(cardCursor
-                        .getColumnIndex(CardDbAdapter.KEY_LOYALTY));
-                card.mText = cardCursor.getString(cardCursor
-                        .getColumnIndex(CardDbAdapter.KEY_ABILITY));
-                card.mFlavor = cardCursor.getString(cardCursor
-                        .getColumnIndex(CardDbAdapter.KEY_FLAVOR));
-            }
+                the next line is a lie. it can start with "SB:"
+
+                a card definition always starts with a number, no formatting (4, 50, 5000) and is
+                always an integer
+
+                then there are two fields that can be swapped:
 
-            /* Override choice is the card can't be foil */
-            if (!CardDbAdapter.canBeFoil(card.setCode, database)) {
-                card.foil = false;
+                card name. these can have spaces in them, so splitting by a space is not possible
+
+                card set. this is encapsulated in []. This doesn't HAVE to be there
+
+                 */
+                if (Character.isDigit(line.charAt(0)) || line.startsWith("SB:")) {
+                    boolean isSideboard = false;
+                    int numberOf;
+                    String cardName = null;
+                    String setName = null;
+                    if (line.startsWith("SB:")) {
+                        isSideboard = true;
+                        line = line.substring(3).trim();
+                    }
+                    int firstSpace = line.indexOf(' ');
+                    numberOf = Integer.valueOf(line.substring(0, firstSpace).trim());
+                    line = line.substring(firstSpace).trim();
+                    int bracketIndex = line.indexOf('[');
+                    if (bracketIndex >= 0) {
+                        int endBracket = line.indexOf(']');
+                        setName = line.substring(bracketIndex + 1, line.indexOf(']'));
+                        if (bracketIndex > 0) {
+                            cardName = line.substring(0, bracketIndex).trim();
+                        } else {
+                            cardName = line.substring(endBracket + 1).trim();
+                        }
+                    }
+                }
             }
-            /* clean up */
-            cardCursor.close();
-            /* return our made card! */
-            return card;
-        } catch (FamiliarDbException | NumberFormatException fde) {
-            /* todo: handle this */
+            return deckList;
+        } catch (FileNotFoundException fnf) {
+            return null; // todo: Clarify what this value should be
```

Comment:
What should these catches return?

An empty list (or whatever list we have) to quell whatever calls this (and reporting the issue to the user)? Or should it return null and be handled on call?